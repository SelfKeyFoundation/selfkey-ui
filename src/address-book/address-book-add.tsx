import * as React from 'react';
import injectSheet, { StyleSheet, WithStyles } from 'react-jss';
import { Grid } from '@material-ui/core';
import StyledButton from '../common/styled-button';

export const styles: StyleSheet = {
    errorText: {
        height: '19px',
        width: '242px',
        color: '#FE4B61',
        fontFamily: 'Lato',
        fontSize: '13px',
        lineHeight: '19px'
    },

    errorColor: {
        color: '#FE4B61 !important',
        border: '2px solid #FE4B61 !important',
        backgroundColor: 'rgba(255,46,99,0.09) !important'
    },

    input: {
        boxSizing: 'border-box',
        height: '46px',
        width: '722px', 
        border: '1px solid #384656',
        borderRadius: '4px',
        backgroundColor: '#1E262E',
        color: '#a9c5d6',
        fontSize: '14px',
        boxShadow: 'inset -1px 0 0 0 rgba(0,0,0,0.24), 1px 0 0 0 rgba(118,128,147,0.2), 2px 0 2px 0 rgba(0,0,0,0.2)',
        paddingLeft: '10px',
        '&::-webkit-input-placeholder': {
            fontSize: '14px',
            color: '#93B0C1',
        }
    },

    inputError: {
        borderBottom: '2px solid #FE4B61 !important'
    },

    label: {
        color: '#93A4AF',
        fontSize: '12px',
        fontWeight: 'bold',
        lineHeight: '15px'
    }

};

export type AddressBookAddState = {
  label: string,
  address: string
};

export type AddressBookAddProps = {
    onSave: (label: string, address: string) => void,
    labelError?: string,
    addressError?: string,
    onCancel: () => void,
    onLabelChange: (label: string) => void,
    onAddressChange: (address: string) => void
}

export type StyledProps = WithStyles<keyof typeof styles> & AddressBookAddProps;

export class AddressBookAddComponent extends React.Component<StyledProps, AddressBookAddState> {

  constructor(props: StyledProps) {
    super(props);
    this.state = {label: '', address: ''};
  }

  handleLabelChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    const label = event.target.value;
    this.setState({...this.state, label});
    this.props.onLabelChange(label);
  };

  handleAddressChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    const address = event.target.value;
    this.setState({...this.state, address});
    this.props.onAddressChange(address);
  };

  handleSubmit = (event: any) => {
    event.preventDefault();
    return this.props.onSave(this.state.label, this.state.address);
  }

  render() {
    const { classes, labelError, addressError, onCancel} = this.props;
    const hasLabelError = (labelError !== '' && labelError !== undefined)
    const hasAddressError = (addressError !== '' && addressError !== undefined)
    const labelInputClass = `${classes.input} ${hasLabelError ? classes.errorColor : ''}`;
    const addressInputClass = `${classes.input} ${hasAddressError ? classes.errorColor : ''}`;
    return (
        <form className={classes.container} id="formwrap" noValidate autoComplete="off" onSubmit={this.handleSubmit}>
            <Grid container direction='column' spacing={32}>
                <Grid item>
                    <Grid container direction='column' spacing={8}>
                        <Grid item>
                            <label className={classes.label}>LABEL</label>
                        </Grid>
                        <Grid item>
                            <input type='text' id='labelInput' onChange={this.handleLabelChange} value={this.state.label} className={labelInputClass} placeholder="Address label" />
                            {hasLabelError &&
                                <span  id='labelError' className={classes.errorText}>{labelError}</span>
                            }
                        </Grid>
                    </Grid>
                </Grid>
                <Grid item>
                    <Grid container direction='column' spacing={8}>
                        <Grid item>
                            <label className={classes.label}>ETH ADDRESS</label>
                        </Grid>
                        <Grid item>
                            <input type='text' id='addressInput' onChange={this.handleAddressChange} value={this.state.address} className={addressInputClass} placeholder="0x" />
                            {hasAddressError &&
                                <span id='addressError' className={classes.errorText}>{addressError}</span>
                            }
                        </Grid>
                    </Grid>
                </Grid>
                <Grid item>
                    <Grid container direction='row' spacing={24}>
                        <Grid item>
                            <StyledButton id='saveButton' variant="contained" size="medium" type="submit" disabled={(!this.state.label || !this.state.address || hasAddressError || hasLabelError)}>
                                Save
                            </StyledButton>
                        </Grid>
                        <Grid item>
                            <StyledButton id='cancelButton' variant="outlined" size="medium" onClick={onCancel}>
                                Cancel
                            </StyledButton>
                        </Grid>
                    </Grid>
                </Grid>
            </Grid>
            
            
        </form>
    );
  }
}

export const AddressBookAdd = injectSheet(styles)(AddressBookAddComponent);

export default AddressBookAdd;
