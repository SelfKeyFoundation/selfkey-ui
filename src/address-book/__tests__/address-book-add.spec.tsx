import * as React from 'react';
import { create } from 'react-test-renderer';
import { mount } from 'enzyme';
import { AddressBookAdd } from '../address-book-add';

describe('Address Book Add', () => {
	it('should show Address Book Add', () => {
		const component = create(
			<AddressBookAdd
				onSave={(label, address) => alert('onEdit' + label + address)}
				onCancel={() => alert('onCancel')}
				onLabelChange={label => alert('onLabelChange ' + label)}
				onAddressChange={address => alert('onAddressChange ' + address)}
				labelError="hey"
			/>
		);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('should defined onChange', () => {
		const component = mount(
			<AddressBookAdd
				onSave={(label, address) => console.log('onEdit' + label + address)}
				onCancel={() => console.log('onCancel')}
				onLabelChange={label => console.log('onLabelChange ' + label)}
				onAddressChange={address => console.log('onAddressChange ' + address)}
				labelError="hey"
			/>
		);
		component.find('#labelInput').simulate('change', { target: { value: '' } });
		component.find('#formwrap').simulate('submit');

		component.find('#addressInput').simulate('change', { target: { value: '' } });
	});
});
