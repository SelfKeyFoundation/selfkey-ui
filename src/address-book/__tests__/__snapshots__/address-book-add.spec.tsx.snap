// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Address Book Add should show Address Book Add 1`] = `
<form
  autoComplete="off"
  id="formwrap"
  noValidate={true}
  onSubmit={[Function]}
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column"
  >
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-direction-xs-column"
      >
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <label
            className="AddressBookAddComponent-label-0-2-5"
          >
            LABEL
          </label>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <input
            className="AddressBookAddComponent-input-0-2-3 AddressBookAddComponent-errorColor-0-2-2"
            id="labelInput"
            onChange={[Function]}
            placeholder="Address label"
            type="text"
            value=""
          />
          <span
            className="AddressBookAddComponent-errorText-0-2-1"
            id="labelError"
          >
            hey
          </span>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-direction-xs-column"
      >
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <label
            className="AddressBookAddComponent-label-0-2-5"
          >
            ETH ADDRESS
          </label>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <input
            className="AddressBookAddComponent-input-0-2-3 "
            id="addressInput"
            onChange={[Function]}
            placeholder="0x"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
      >
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <button
            className="MuiButtonBase-root MuiButton-root Component-root-0-2-6 MuiButton-contained MuiButton-containedPrimary Component-containedPrimary-0-2-7 Mui-disabled Component-disabled-0-2-9 Mui-disabled"
            disabled={true}
            id="saveButton"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={-1}
            type="submit"
          >
            <span
              className="MuiButton-label"
            >
              Save
            </span>
          </button>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <button
            className="MuiButtonBase-root MuiButton-root Component-root-0-2-6 MuiButton-outlined Component-outlined-0-2-8 MuiButton-outlinedPrimary"
            disabled={false}
            id="cancelButton"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              Cancel
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</form>
`;
